Add FS_MOUNT_FLAG_USE_DISK_ACCESS to fs_mount flags

de-nordic opened the pull request:
The commit adds new mount flag: FS_MOUNT_FLAG_USE_DISK_ACCESS that request using Disk Access API, when a file system driver supports it. File systems should by default use Flash API, but in case when they support Disk Access API, this flag makes them choose that API.
In case when File systems, like for example FAT, only support the DIsk Access API, then existence of the flag does not matter, but the file system, upon mount, should set the flag to indicate that it is using the Disk Access API.
When a file system does not support the flag and it is set, then fs_mount of the file system should return -ENOSUP.

de-nordic committed:
fs: Add FS_MOUNT_FLAG_USE_DISK_ACCESS flag

The commit adds new mount flag that requests, when supported by
a file system driver, to use Disk Access API.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
fs: fat_fs: Set FS_MOUNT_FLAG_USE_DISK_ACCESS on mount

Mounting FAT FS will set FS_MOUNT_FLAG_USE_DISK_ACCESS to the flags
of an object serving the mount point.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic committed:
fs: littlefs: Add processing of FS_MOUNT_FLAG_USE_DISK_ACCESS flag

Atempting to mount LittleFS with the flag will cause the fs_mount
to return -ENOSUP and log error message.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

de-nordic commented:
@lmajewski FYI.

Laczen commented:
@de-nordic, could you remove the `mgmt/mcumgr/lib` commit. I like the idea of putting this in a flag.

de-nordic commented:
> @de-nordic, could you remove the `mgmt/mcumgr/lib` commit. I like the idea of putting this in a flag.

@Laczen sorry, my bad. Will remove.

Laczen commented on include/fs/fs.h:
```diff
@@ -66,7 +66,14 @@ enum {
  */
 #define FS_MOUNT_FLAG_AUTOMOUNT BIT(2)
-
+/** Flag requests file system to use Disk Access API. When a file system supports
```
Flag to force the file system to use the Disk Access API. When this flag is
set prior to fs_mount the file system should use the Disk Access API. When the flag is not set prior to fs_mount the file system can use its preferred API.
When a file system only supports the Disk Access API it should set the flag.
`

de-nordic replied:
Little bit reworded your suggestion, but generally followed the comment.

Laczen replied:
> 

NP, looks good to me.

de-nordic committed:
dts: Add "disk-access" property to fstab bindings

The commit adds "disk-access" property to fstab bindings to allow
selecting FS_MOUNT_FLAG_USE_DISK_ACCESS while defining file
systems in DTS.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

mbolivar-nordic commented on dts/bindings/fs/zephyr,fstab-common.yaml:
```diff
@@ -41,3 +41,11 @@ properties:
+  disk-access:
+    type: boolean
+    description: |
+      Use disk-access for accessig storage media.
```
```suggestion
      Use disk-access for accessing storage media.
```

de-nordic replied:
Done
